index.js

```diff
@@ -1,68 +1,79 @@
-    removeBtn.addEventListener('click', () => removeBook(book));
+      removeBtn.addEventListener('click', () => this.removeBook(book));
 
-    bookTitle.innerHTML = book.title;
-    bookAuthor.innerHTML = book.author;
-    removeBtn.innerHTML = 'Remove';
+      bookTitle.innerHTML = book.title;
+      bookAuthor.innerHTML = book.author;
+      removeBtn.innerHTML = 'Remove';
 
-    bookContainer.appendChild(bookTitle);
-    bookContainer.appendChild(bookAuthor);
-    bookContainer.appendChild(removeBtn);
-    booksContainer.appendChild(bookContainer);
-  });
-}
-populateBooks();
+      bookContainer.appendChild(bookTitle);
+      bookContainer.appendChild(bookAuthor);
+      bookContainer.appendChild(removeBtn);
+      booksContainer.appendChild(bookContainer);
+    });
+  }
 
-function removeBook(item) {
-  books = books.filter((book) => book.id !== item.id);
-  booksContainer.innerHTML = '';
-  preserveBookShelf();
-  populateBooks();
-}
+  static addBook() {
+    const addBookForm = document.querySelector('#add-book');
+    addBookForm.addEventListener('submit', (event) => {
+      event.preventDefault();
+      const title = document.querySelector('#title').value;
+      const author = document.querySelector('#author').value;
+      if (title !== '' && author !== '') {
+        const uniqueId = document.querySelector('#title').value + Math.floor(Math.random() * 1000);
+        const newBook = new Book(document.querySelector('#title').value, document.querySelector('#author').value, uniqueId);
+        books.push(newBook);
+        document.getElementById('books').innerHTML = '';
 
-addBookForm.addEventListener('submit', (event) => {
-  event.preventDefault();
-  const title = document.querySelector('#title').value;
-  const author = document.querySelector('#author').value;
-  if (title !== '' && author !== '') {
-    const uniqueId = document.querySelector('#title').value + Math.floor(Math.random() * 1000);
-    books.push({ id: uniqueId, title: document.querySelector('#title').value, author: document.querySelector('#author').value });
-    booksContainer.innerHTML = '';
-    preserveBookShelf();
-    populateBooks();
-    document.querySelector('#title').value = '';
-    document.querySelector('#author').value = '';
-  } else {
-    alert('Please enter title and/or author');
+        this.bookLocalStorage();
+        this.displayBooks();
+        document.querySelector('#title').value = '';
+        document.querySelector('#author').value = '';
+      } else {
+        alert('Please enter title and/or author');
+      }
+    });
   }
-});
+
+  static removeBook(item) {
+    books = books.filter((book) => book.id !== item.id);
+    document.getElementById('books').innerHTML = '';
+    this.bookLocalStorage();
+    this.displayBooks();
+  }
+
+  static bookLocalStorage() {
+    let bookShelf = '';
+    bookShelf = JSON.stringify(books);
+    window.localStorage.setItem('bookShelf', bookShelf);
+  }
+}
+
+Book.addBook();
+Book.displayBooks();
```

Comment:
- [ ] You have done a great job so far 👏 , however, it seems that you did not initialize the books array inside the constructor method as mentioned in the project requirements below 👇 
![image](https://user-images.githubusercontent.com/55465133/137993798-bc364cc8-62ec-482c-99ca-bc26a19891c3.png)

- [ ] It seems that you have mixed responsibilities, the class should only be there to manage the `books` array but not to take the responsibility of manipulating the DOM. Ideally, it should only have like 3 methods: `addBook, removeBook, getBooks` then the `books` array. Then you can create a single instance of the class for you to access all these methods when you are manipulating the DOM. this means when for example you submit the form for adding a new book, you should do something like this 👇 
```
const book = new Book(); // create a single and unique instance of the class
form.addEventListiner('submit', (e) => {
    e.preventDefault();
    book.addBook({ // then call the addBook method from the class to add a new book for you
       author: .... 
       title: ....
    })
})
```
As you can see with this, you don't need to manipulate the DOM inside the class 😉 

- [ ] [OPTIONAL] Also I don't think it's a good idea to mark all the methods in the class as `static` if you are not using them directly from outside ex: `bookLocalStorage, removeBook` there is no need for these methods to be static since you only use the inside the class. 👍

Reply:
This issue is well resolved! ✔️